homeassistant/helpers/schema_config_entry_flow.py

```diff
@@ -128,30 +130,24 @@ async def _async_form_step(
                     ):
                         user_input[str(key.schema)] = key.default()
 
-        if user_input is not None and form_step.validate_user_input is not None:
+        if form_step.validate_user_input is not None:
             # Do extra validation of user input
             try:
                 user_input = form_step.validate_user_input(self, user_input)
             except SchemaFlowError as exc:
                 return self._show_next_step(step_id, exc, user_input)
 
-        if user_input is not None:
-            # User input was validated successfully, update options
-            self._options.update(user_input)
-
-        next_step_id: str = step_id
-        if user_input is not None or form_step.schema is None:
```

Comment:
Since `user_input` was never `None`, checking `form_step.schema` was pointless.

The check will be added again in a different PR.